bolt-design-system/bolt
# Fix Blockquote content disappearing

## Jira

- http://vjira2:8080/browse/BDS-1274
- http://vjira2:8080/browse/BDS-970

## Summary

- Fix IE bug where content disappears after JS loads + adds Jest and Nightwatch tests (via #1168) to help prevent this bug from reoccurring.

## Details
Use of `bolt-text` as slotted content seems to be causing this bug in IE, really anywhere `bolt-blockquote` is rendered to the light DOM. For now, I replaced `bolt-text` with the HTML generated by `bolt-text`. This is a workaround until we can fix the underlying issue, which seems to be related to shadyDOM and/or nesting slotted content.

Part of the problem was also our mutation observer getting triggered multiple times when watching chilidNodes in the light DOM. For now, I've turned this off when rendering to the light DOM.

Lastly, I added tests to catch any regressions like this in the future.

## How to test
- Review the [files changed](https://github.com/bolt-design-system/bolt/pull/1167/files).
- View the [Blockquote demo page](https://fix-vr-blockquote-content-disappearing.boltdesignsystem.com/pattern-lab/patterns/02-components-blockquote/index.html) in IE Edge and IE11 (with JS on) and verify the Blockquote text and images do not disappear after the page loads.
